File: unorganised/P664.c
#include <stdio.h>
#include <stdlib.h>

static int nextInt() {
	int nums;
	int prim = getchar_unlocked() - '0';
	while ((nums = getchar_unlocked()) >= '0')
		prim = 10*prim + nums-'0';

	return prim;
}

int main() {
	int p, j, dev, mem, n, i;
	int alt[5];
	while(1) {
		p = nextInt();
		if(p == 0) break;
		mem = 0;
		for(i = 0; i < p; i++) {
			dev = 0;
			n = nextInt();
			for(j = 0; j < n; j++) {
				alt[j] = nextInt();
				if(alt[j] > dev) {
					dev = alt[j];
				}
			}
			if(i == 0 || mem > dev) {
				mem = dev;
			}
		}
		printf("%d\n", mem);

	}
}

File: unorganised/P664.java
package aop.examen;

import java.io.BufferedInputStream;
import java.util.Scanner;

public class P664 {

	public static void main(String[] args) {
		try (Scanner sc = new Scanner(new BufferedInputStream(System.in, 1 * 1024))) {
			int p = 1;
			while(true) {
				p = Integer.parseInt(sc.nextLine());
				if(p == 0) break;
				int dev = 0;
				int mem = 0;
				for(int i = 0; i < p; i++) {
					String alt1 = sc.nextLine();
					String[] alt = alt1.split(" ");
					for(int j = 1; j < alt.length; j++) {
						if(Integer.parseInt(alt[j]) > dev) {
							dev = Integer.parseInt(alt[j]);
						}
					}
					if(i == 0) {
						mem = dev;
					} else {
						if(mem > dev) {
							mem = dev;
						}
					}
					dev = 0;
				}
				System.out.println(mem);
				
			}
		} catch (Exception e) {
			System.out.println("error");
			System.exit(1);
		}


	}

}
